File: code/dynamic5.java
import java.lang.reflect.Array;
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) arr[i] = sc.nextInt();
        int []table = new int[n];
        Arrays.fill(table,0);
        table[0] =1;
        int cur = arr[0];
        for( int i=1;i<n;i++){
            if(cur<arr[i]) table[i] =table[i-1]+1;
            else table[i] = table[i-1];
            cur = table[i];
        }
        System.out.println(n-table[-1]);
    }

}

File: code/dynamic5.py
def solution(arr):
    table =[0]*len(arr)
    table[0] =1
    cur = arr[0]
    for i in range(1,len(arr)):
        if cur>arr[i]: table[i]=table[i-1]+1
        else: table[i] =table[i-1]
        cur= arr[i]
    return len(arr) - table[-1]

arr=[2,7,5,2]
print(solution(arr))
